15-connect-app-and-store/README.md

```diff
@@ -0,0 +1,527 @@
+```
+Next, lets add a `constructor()` in `<App />` which will handle creating the store.
+
+```js
+// App.js
+
+constructor(props) {
+  //It is always necessary to call super(props)
+  //whenever adding functionality to the constructor
+  super(props);
+
+  this._store = createStore(
+    emails,
+    applyMiddleware(thunk)
+  );
+}
+```
+We have a `store`! Now, that we do, we can modify our `render()` function so it returns `<Provider />` as the top level component, with `<Page />` as its child.
+
+```js
+//App.js
+
+render() {
+  <Provider store={this._store}>
+    <Page />
+  </Provider>
+}
+```
+
+After these changes our App.js should look something like:
+
+```js
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {createStore, applyMiddleware} from 'redux';
+import thunk from 'redux-thunk';
+import {Provider} from 'react-redux';
+
+import Page from './containers/Page';
+
+import {emails} from './reducers';
+
+export default class App extends PureComponent {
+  static propTypes = {
+    pollInterval: PropTypes.number
+  };
+
+  static defaultProps = {
+    // default the `pollInterval` prop to 2 secs when not specified
+    pollInterval: 2000
+  };
+
+
+  constructor(props) {
+    super(props);
+
+    this._store = createStore(
+      emails,
+      applyMiddleware(thunk)
+    );
+  }
+
+  componentDidMount() {
+    // Retrieve emails from server once we know DOM exists
+    this._getUpdateEmails();
+
+    // Set up long-polling to continuously get new data
+    this._pollId = setInterval(
+      () => this._getUpdateEmails(),
+      this.props.pollInterval
+    );
+  }
+
+  componentWillUnmount() {
+    // Need to remember to clearInterval when the component gets
+    // removed from the DOM, otherwise the interval will keep going
+    // forever and leak memory
+    clearInterval(this._pollId);
+  }
+
+  _getUpdateEmails() {
+    return getEmails().then(emails => this.setState({emails}))
+  }
+
+  render() {
+    return (
+      <Provider store={this._store}>
+        <Page />
+      </Provider>
+    );
+  }
+}
+```
+There is one last thing we need to update before we can move on to updating `<Page/>`: hydration. We are still calling `getEmails` inside `_getUpdateEmails` and then updating our `state`! We instead want to call the Redux action we created in [Step 14](../14-reduxy-actions-reducers) and let the action handle hydrating the store. First, lets import it and remove the import from `'./action-reducers'`:
+```js
+// App.js
+
+//other imports
+import {getEmails} from '../actions';
+```
+With that we are good to go right? Nope! We need to wrap the function in a call to `dispatch()` so our "action creators" know to listen for it. Since `dispatch()` is a property of the `store` we can access it and call it directly. We can do that by modifying `getUpdateEmails()` in the following way:
+```js
+//App.js
+
+_getUpdateEmails() {
+  return this._store.dispatch(getEmails());
```

Comment:
I think once this moves into `Page` like I suggested above, then `getEmails()` will be "connected" so you don't have to dispatch it manually on the store. Manual dispatches is atypical